File: competitions/101-Hack-52/construct-the-array.py
#https://www.hackerrank.com/contests/101hack52/challenges/construct-the-array

MOD = 1000000007

if __name__ == "__main__":
    n, k, x = [int(_) for _ in input().strip().split()]
    if x == 1:
        p = k-1
    else:
        p = k-2
    total = k-1
    for i in range(2,n-1):
        total = (total * (k-1)) % MOD
        p = (total - p) % MOD
    print(p)

File: competitions/101-Hack-52/construct-the-array.cpp
#include<iostream>
using namespace std;

int MOD = 1000000007;

int main() {
	int n, x, k;
	long p;
	cin >> n >> k >> x;
	if (x == 1) {
		p = k-1;
	}
	else {
		p = k-2;
	}
	long total = k-1;
	for (int i=2; i<n-1; i++) {
		total = (total * (k-1)) % MOD;
		p = (total - p) % MOD;
	}
	cout << p;
	return 0;
}
